tests/tests/operation_tests.cpp

```diff
@@ -840,6 +840,70 @@ BOOST_AUTO_TEST_CASE( prediction_market_resolves_to_0 )
    }
 }
 
+/***
+ * Prediction markets should not suffer a black swan (Issue #460)
+ */
+BOOST_AUTO_TEST_CASE( prediction_market_black_swan )
+{ 
+   try {
+      ACTORS((judge)(dan)(nathan));
+
+      // progress to recent hardfork
+      generate_blocks( HARDFORK_CORE_1270_TIME );
+      set_expiration( db, trx );
+
+      const auto& pmark = create_prediction_market("PMARK", judge_id);
+
+      int64_t init_balance(1000000);
+      transfer(committee_account, judge_id, asset(init_balance));
+      transfer(committee_account, dan_id, asset(init_balance));
+
+      update_feed_producers( pmark, { judge_id });
+      price_feed feed;
+      feed.settlement_price = asset( 1, pmark.id ) / asset( 1 );
+      publish_feed( pmark, judge, feed );
+
+      borrow( dan, pmark.amount(1000), asset(1000) );
+
+      // feed a price that will cause a black swan
+      feed.settlement_price = asset( 1, pmark.id ) / asset( 1000 );
+      publish_feed( pmark, judge, feed );
+
+      // verify a black swan happened
+      GRAPHENE_REQUIRE_THROW(borrow( dan, pmark.amount(1000), asset(1000) ), fc::exception);
+
+      // interestingly, PMARKII cannot be created without dan force_settling
+      force_settle( dan, pmark.amount(1000) );
```

Comment:
Without this, `judge` cannot make another prediction market. This may be normal, but seems strange. I am investigating. Please comment if there is a good reason for this.

Reply:
Cannot? It's not normal. What's the error.

Reply:
```
2469515ms th_a       operation_tests.cpp:902       test_method          ] e.to_detail_string(): 10 assert_exception: Assert Exception
!_bitasset_data->has_settlement(): Cannot update debt position when the asset has been globally settled
    {}
    th_a  market_evaluator.cpp:219 do_evaluate
```
I will step through to follow the logic.

Reply:
My fault. I forgot to trx.clear().